[Feature] Video support banner

kimberleyworld opened the pull request:
This is a work in progress. Added and styled video to support bar.

kimberleyworld committed:
Bringing in form and new meta data

Brought across the support form with no styles. Added a headings field to the options-fundraising and changed repeatable to false for the copy.

kimberleyworld committed:
Updated grid layout

Got rid of some extra divs and ensured layout is responsive.

kimberleyworld committed:
Spaccing

Adjusted padding and spacing

varffs commented on lib/theme-options/options-fundraising.php:
```diff
@@ -222,11 +222,17 @@ function nm_register_fundraising_options_metabox() {
 
   $main_options->add_field( array(
+    'name' => __( 'CTA Headline', 'NM' ),
+    'desc' => 'Add headline.',
```
lets be more descriptive for end users. Explain what happens when this isn't set: that there is a fallback from the normal support section title

varffs commented on partials/specials/banners/support-video.php:
```diff
@@ -1,43 +1,114 @@
+                
+                <!-- Support section text or custom message -->
+                <?php
+                  if ($support_section_text || $cutomText) {
```
i think we can refactor this as we should never have a totally empty moment. In the initial variable declarations lets check for our first option, and if not set check for our second, and if thats not set fallback to hardcoded text taken from the site as it currently is. Then in this location we dont need to validate content just echo

kimberleyworld replied:
I have applied same thinking to the you tube ID now that we have added a fallback ID from page-support and a hard coded value.

varffs commented on partials/specials/banners/support-video.php:
```diff
@@ -1,43 +1,114 @@
+                  }
+                ?>
+                
+                <!-- Donation amount options -->
```
lets work out how to turn the form rendering into a helper function so we can duplicate this from here and also the support section file. Probably for now it can just be a new renderer function with no parameters that outputs all this form (or maybe it needs the autovalues variable, but that data could be got inside the renderer function)

kimberleyworld committed:
Resolved comments

- Added renderer function to create the form
- exicuted it in suport-video.php and support-section.php
- added more detail to user input fields WP side

kimberleyworld commented:
I have made the updates we spoke about. Let me know if they look ok?

varffs commented on lib/renderers.php:
```diff
@@ -7,44 +61,47 @@
  * @param integer  $number_of_posts The number of posts to render
  */
-function render_see_also($query, $number_of_posts = 1) {
+function render_see_also($query, $number_of_posts = 1)
```
careful with autolinting and git. you can end up with diffs that are hard to read because you've changed some of the underlying code and a lot of formatting. This file isn't too bad because the new code is clear, but the changes to the existing codebase would be better to avoid committing and to do in a separate branch. 

In regards to PHP linting this is def something we should do, which config file/plugin is doing this linting? we need to ensure we have shared config files in the repo so that everyone lints the same way.

kimberleyworld replied:
I have set my linter to format on save, which would have changed the entire file. I need to work out how to ensure that it's only running for new code or files.  I am using the elephant linting plug in but lets compare. 

I'll get more info on this when we chat.

varffs commented on partials/specials/banners/support-video.php:
```diff
@@ -1,43 +1,73 @@
+// Get options and values
+$support_section_text = NM_get_option('nm_fundraising_settings_support_section_text', 'nm_fundraising_options');
+$heading_copy = !empty($args['heading_copy']) ? $args['heading_copy'] : NM_get_option('nm_fundraising_settings_support_section_title', 'nm_fundraising_options');
+$customYoutube_id = NM_get_option('nm_fundraising_settings_video_banner_cta_youtube_id', 'nm_fundraising_options');
```
we need a consistent php variable format. we should use snake case as that is the existing code convention that I've been working towards (and I think it is the most readable). I will document this in a code style guide document somewhere

kimberleyworld replied:
noted - swopped to $custom_text and $custom_headline

varffs commented on partials/specials/banners/support-video.php:
```diff
@@ -1,43 +1,73 @@
-    </div>
-    <div class="grid-row">
-      <div class="grid-item offset-s-0 is-s-24 offset-m-2 is-m-20 is-xl-12 is-xxl-14 mb-s-3">
+<div id="<?php echo $instance; ?>" class="support-section background-red font-color-white pt-2 pb-2 pt-s-2 pb-s-2 pl-0 pt-s-5 pb-s-5 ">
```
`$instance` isnt defined here so isn't outputting anything. the `support-section` class is used as a hook for the javascript to initiate the form. this doesnt seem to be breaking anything, but it shouldn't be declared here and then again inside. should just be declared on the form itself. (actually probably needs to be renamed because that class isn't really descriptive enough. should be more like `ux-support-form` or something like that)

varffs replied:
worth reading the `Support.js` file to understand the mechanism (and consider how it possibly could be refactored if needed)

kimberleyworld replied:
I see what you mean, i have removed the support-section class from out side the render function and instead added it onto the form element which should work as intended and call the correct functions from what I understand in the support.js file. 
  <form class=" support-section support-form" action="https://donate.novaramedia.com/regular"> 

I have moved id="<?php echo $instance; ?>" into the render function so the $instance remains in scope. I see no reason for it to be needed outside of the form element.

varffs commented on partials/specials/banners/support-video.php:
```diff
@@ -1,43 +1,73 @@
       </div>
-      <div class="grid-item offset-s-0 is-s-24 offset-m-2 is-m-20 is-xl-12 flex-item-xxl-4">
+
+      <!-- Title and form section -->
```
would like to avoid public visible code comments in the final output as much as possible. Totally fine and good to comment inside PHP so it isn't rendered in the final markup. We could work comment removal into build scripts but think thats overkill.

varffs commented on partials/specials/banners/support-video.php:
```diff
@@ -1,43 +1,73 @@
+              if ($customHeadline) {
+                echo $customHeadline;
+              } else if ($heading_copy) {
+                echo $heading_copy;
```
would like more descriptive variable names here. I think `$heading_copy` should be `$default_support_section_heading` or similar

kimberleyworld replied:
changed to $default_support_section_text and $default_support_section_heading

varffs commented on partials/specials/banners/support-video.php:
```diff
@@ -1,43 +1,73 @@
+          </div>
         </a>
-        <a href="<?php echo site_url('support/'); ?>" class="ui-button ui-button--white">Fund something better</a>
+        <!-- Reneders the support form -->
```
there is some layout issue around this code. The grid isn't correct in the final output. more detail with screenshots

kimberleyworld replied:
I believe I have fixed the layout issues but have noticed that my fonts are different and need to look at the styles files more closely. I have added it in a to do list for us to go through to check I have installed properly.

varffs reviewed (changes requested):
**Im doing a very in depth review here so we catch anything and debug from the start. Just to make clear that yes I am nit picking and sorry for being perhaps annoying**

There are a few details in the code. Mostly in style. I'm going to create a document with human readable style guide for the languages that we can try to align with any code linting tools etc.

The main issue however is that the output layout isn't correct. The previous support section doesn't render the same as before:

<img width="1425" alt="Screenshot 2024-11-15 at 12 02 49" src="https://github.com/user-attachments/assets/9e1c7f4f-fe53-471f-a1c6-c5576b1983b6">

And you can see the same layout issues on the video section:

<img width="1502" alt="Screenshot 2024-11-15 at 12 02 33" src="https://github.com/user-attachments/assets/d54d270c-29dd-4573-b940-f750c1cb9cfb">

I appreciate that there isn't a fully spec'd design but the video support section needs to match the previous support section as closely as possible in regards to margins and padding too. So at the moment there isn't the same outside padding on the element (but the usage of helper classes is correct just needs more padding)

There is also an issue whereby with the existing data the video section text is rendering as `Array` which means that some old metadata format isn't working with the new approach. This is something always to try to catch so that when we push changes live we dont get strange outputs publicly. Sometimes it is totally acceptable to make sure that the data is updated at the exact moment of pushing changes live. But often it is better to either use a totally new meta key (this feature would be a good moment for that) or make sure to validate and fix the data that you get from metadata requests.

kimberleyworld committed:
Going through pull request changes

- adjusted formatting, padding and margins
- added new meta key by changing ID
- changed variable format to snake case
- support-section class and $instance definition into rendered function
- moved comments inside php
- added more descriptive variable names in support-video.php

kimberleyworld committed:
Updated comments

Made comments more descriptive inline with documentation

varffs commented on partials/specials/banners/support-video.php:
```diff
@@ -1,21 +1,18 @@
-$customHeadline = NM_get_option('nm_fundraising_settings_video_banner_cta_headline', 'nm_fundraising_options');
+$custom_text = NM_get_option('nm_fundraising_settings_video_banner_cta_custom_text', 'nm_fundraising_options');
+$custom_headline = NM_get_option('nm_fundraising_settings_video_banner_cta_headline', 'nm_fundraising_options');
 $youtube_id = get_post_meta($post_id, '_nm_support_youtube', true);
```
There is never any `$post_id` declared from the Support page in any way. So I don't think this line could work. It also should have a more descriptive variable name

varffs commented on partials/specials/banners/support-video.php:
```diff
@@ -1,21 +1,18 @@
-$heading_copy = !empty($args['heading_copy']) ? $args['heading_copy'] : NM_get_option('nm_fundraising_settings_support_section_title', 'nm_fundraising_options');
+$default_support_section_text = NM_get_option('nm_fundraising_settings_support_section_text', 'nm_fundraising_options');
+$default_support_section_heading = !empty($args['heading_copy']) ? $args['heading_copy'] : NM_get_option('nm_fundraising_settings_support_section_title', 'nm_fundraising_options');
 $customYoutube_id = NM_get_option('nm_fundraising_settings_video_banner_cta_youtube_id', 'nm_fundraising_options');
```
missed snake case variable name

varffs reviewed (changes requested):
Other than the minor data source for the support page video I think this is ready for a look over by design eyes. I think we will certainly want to make some changes to the way it is laid out over the responsive breakpoints, but that is trivial design stuff to change.

*edit: Also as I had forgotten. We need to update the change log for this feature

kimberleyworld committed:
Support page youtube id and snake case

kimberleyworld committed:
Update CHANGELOG.md

deepsource-autofix[bot] committed:
style: format code with Prettier

This commit fixes the style issues introduced in 48bf222 according to the output
from Prettier.

Details: https://github.com/novaramedia/novaramedia-com/pull/311

kimberleyworld committed:
Merge branch 'tooling-phpcs-config' into feature_videoSupportBar

kimberleyworld committed:
linting corrections

kimberleyworld committed:
Design review changes

- video full width when on two rows
- video and form 50/50 large screens
- removed margin top on heading except for L screens (2 columns)

varffs committed:
Revert "Merge branch 'tooling-phpcs-config' into feature_videoSupportBar"

This reverts commit ecca1a6a1459080a85cd03cb0b0a4498e62ea645, reversing
changes made to 308d283ee6130e4ba84355aa32617bde3c11c05a.

varffs committed:
Merge branch 'development' into feature_videoSupportBar

kimberleyworld committed:
Formatted relevant parts

kimberleyworld committed:
Updated support video

kimberleyworld committed:
Update options-fundraising.php

kimberleyworld committed:
changed webpack version
